high_level/profile/profile_creation.md

```diff
@@ -0,0 +1,112 @@
+The profile folder is stored in a separate place from the Firefox program so that, if something ever goes wrong with Firefox, the profile information will still be there.
+
+Firefox also comes with a profile manager, a different run mode to create, migrate and delete the profiles.
+
+## Profile Behaviors
+
+In order to understand the behavior of users and base analysis on things like the profile creation date,
+it is essential to understand how a profile is created and identified by the browser.
+
+A profile can also be reset or deleted by the user.
+The user can opt-out of or opt-in to sending Telemetry data.
+
+These different cases are described in more detail in the following sections.
+
+### Profile Creation
+
+#### First use
+
+When Firefox is opened for the first time after a fresh install, without any prior Firefox profile on disk, it will create a new profile.
+Firefox chooses a random name, create the profile directory and marks the new profile as default for subsequent starts of Firefox.
+
+#### Default or Profile Manager creation
+
+On subsequent starts, Firefox tries to find the profile directory and load the profile data (bookmarks, addons, etc.).
+If not otherwise specified it uses the default (or last used) profile, finds the profile directory and starts loading files from there.
+
+The user can create a new profile through the profile manager.
+This can either be done on `about:profiles` or by starting Firefox with the `--ProfileManager` flag.
+The profile manager will ask for a name for the profile and picks a new directory for it.
+The directory can be changed by the user to another new or even an existing directory.
+If that directory does not exist, it is created.
+Additional files for bookmarks are placed inside this directory.
+
+#### Command-line start
+
+Firefox can be started on the command line with a path to a profile directory: `firefox --profile path/to/directory`.
+In that case the passed profile directory is, if not already existing, created. No files inside that directory are written at launch.
+
+### Profile Reset
+
+TODO
+
+### Profile Deletion
+
+A profile can be deleted by simply removing the profile directory.
+We will never know about a deletion. We simply won't see that profile turning up in Telemetry data anymore.
+
+### Telemetry opt-out
+
+TODO
+
+## Profile Creation Date
+
+The *profile creation date* is the assumed date of initial profile creation.
+However it proofed to be not reliable for all cases.
+There are multiple ways this date is determined.
+
+### During Profile Creation
+
+When a profile is created explicitly (default profile on first launch, new profile from Profile Manager or the `--createprofile` on the command line), the profile directory is created and a `times.json` containing a timestamp of the current time is stored inside that profile directory[^1].
+It is read at later times when the profile creation date is used.
```

Comment:
this category of profile creation (created explicitly) seems super useful, would be great if we could define it in the profile creation section above:

created explicity
* default profile on first launch
* new profile from Profile Manager
* --createprofile command

not created explicitly (?)
* --profile

Reply:
it would help understanding the flowchart a lot